packages/program/src/contract-call-script.ts

```diff
@@ -277,40 +282,42 @@ export const getScriptDataV1: ContractCallScriptFn = (
   segmentOffset: number
 ): { scriptData: Uint8Array[]; callParamOffsets: CallOpcodeParamsOffset } => {
   const scriptData: Uint8Array[] = [];
-  const callSegmentOffset = segmentOffset + WORD_SIZE;
-  let gasForwardedSize: number = 0;
+
+  const amountOffset = segmentOffset;
+  const assetIdOffset = amountOffset + WORD_SIZE;
+  const callDataOffset = assetIdOffset + ASSET_ID_LEN;
+  const encodedSelectorOffset = callDataOffset + CONTRACT_ID_LEN + WORD_SIZE + WORD_SIZE;
+  const encodedArgsOffset = encodedSelectorOffset + call.fnSelectorBytes.byteLength;
+  const encodedArgs = arrayify(call.data);
+  let gasForwardedOffset = 0;
 
   // 1. Amount
   scriptData.push(new BigNumberCoder('u64').encode(call.amount || 0));
   // 2. Asset ID
   scriptData.push(new B256Coder().encode(call.assetId?.toString() || BaseAssetId));
-  // 3. Gas to be forwarded
+  // 3. Contract ID
+  scriptData.push(call.contractId.toBytes());
+  // 4. Function selector offset
+  scriptData.push(new BigNumberCoder('u64').encode(encodedSelectorOffset));
+  // 5. Encoded argument offset
+  scriptData.push(new BigNumberCoder('u64').encode(encodedArgsOffset));
+  // 6. Encoded function selector
+  scriptData.push(call.fnSelectorBytes);
+  // 7. Encoded arguments
+  scriptData.push(encodedArgs);
+
+  // 8. Gas to be forwarded
   if (call.gas) {
     scriptData.push(new BigNumberCoder('u64').encode(call.gas));
-    gasForwardedSize = WORD_SIZE;
+    gasForwardedOffset = encodedArgsOffset + encodedArgs.byteLength;
```

Comment:
Gas offset was in the wrong place, thus breaking multicall. Not identified in the previous tests as other offsets were set correctly.